osquery/tests/integration/tables/interface_details.cpp

```diff
@@ -16,57 +16,54 @@
-  // 2. Check size before validation
-  // ASSERT_GE(data.size(), 0ul);
-  // ASSERT_EQ(data.size(), 1ul);
-  // ASSERT_EQ(data.size(), 0ul);
-  // 3. Build validation map
-  // See IntegrationTableTest.cpp for avaialbe flags
-  // Or use custom DataCheck object
-  // ValidatatioMap row_map = {
-  //      {"interface", NormalType}
-  //      {"mac", NormalType}
-  //      {"type", IntType}
-  //      {"mtu", IntType}
-  //      {"metric", IntType}
-  //      {"flags", IntType}
-  //      {"ipackets", IntType}
-  //      {"opackets", IntType}
-  //      {"ibytes", IntType}
-  //      {"obytes", IntType}
-  //      {"ierrors", IntType}
-  //      {"oerrors", IntType}
-  //      {"idrops", IntType}
-  //      {"odrops", IntType}
-  //      {"collisions", IntType}
-  //      {"last_change", IntType}
-  //      {"link_speed", IntType}
-  //      {"pci_slot", NormalType}
-  //      {"friendly_name", NormalType}
-  //      {"description", NormalType}
-  //      {"manufacturer", NormalType}
-  //      {"connection_id", NormalType}
-  //      {"connection_status", NormalType}
-  //      {"enabled", IntType}
-  //      {"physical_adapter", IntType}
-  //      {"speed", IntType}
-  //      {"service", NormalType}
-  //      {"dhcp_enabled", IntType}
-  //      {"dhcp_lease_expires", NormalType}
-  //      {"dhcp_lease_obtained", NormalType}
-  //      {"dhcp_server", NormalType}
-  //      {"dns_domain", NormalType}
-  //      {"dns_domain_suffix_search_order", NormalType}
-  //      {"dns_host_name", NormalType}
-  //      {"dns_server_search_order", NormalType}
-  //}
-  // 4. Perform validation
-  // validate_rows(data, row_map);
+TEST_F(InterfaceDetailsTest, sanity) {
+  QueryData const rows = execute_query("select * from interface_details");
+  auto const row_map = ValidatatioMap{
+      {"interface", NonEmptyString},
+      {"mac", NonEmptyString},
+      {"type", NonNegativeInt},
+      {"mtu", NonNegativeInt},
+      {"metric", IntType},
+      {"flags", NonNegativeInt},
+      {"ipackets", NonNegativeInt},
+      {"opackets", NonNegativeInt},
+      {"ibytes", NonNegativeInt},
+      {"obytes", NonNegativeInt},
+      {"ierrors", NonNegativeInt},
+      {"oerrors", NonNegativeInt},
+      {"idrops", NonNegativeInt},
+      {"odrops", NonNegativeInt},
+      {"collisions", NonNegativeInt},
+      {"last_change", IntType},
+#ifdef OSQUERY_POSIX
+      {"link_speed", IntType},
+#endif
+#ifdef OSQUERY_LINUX
+      {"pci_slot", NormalType},
+#endif
+#ifdef OSQUERY_WINDOWS
+      {"friendly_name", NormalType},
+      {"description", NormalType},
+      {"manufacturer", NormalType},
+      {"connection_id", NormalType},
+      {"connection_status", NormalType},
+      {"enabled", Bool},
+      {"physical_adapter", IntType},
+      {"speed", IntType},
+      {"service", NormalType},
+      {"dhcp_enabled", IntType},
+      {"dhcp_lease_expires", NormalType},
+      {"dhcp_lease_obtained", NormalType},
```

Comment:
NonNegativeInt ?

Reply:
Unfortunately, same reason as for the `"dhcp_lease_expires"`.